src/atmos_param/cloud_simple/large_scale_cloud.F90

```diff
@@ -0,0 +1,334 @@
+            rhc_surf, rhc_top, n_rhc, &
+            cf_diag_formula_name, &
+            linear_a_surf, linear_a_top, linear_power, &
+            slingo_rhc_low, slingo_rhc_mid, slingo_rhc_high, &
+            do_adjust_cld_by_omega, omega_adj_threshold, adj_pres_threshold, &
+            do_freezedry, qv_polar_val, freezedry_power
+
+  contains
+
+  subroutine large_scale_cloud_init(axes, Time)
+    type(time_type), intent(in)       :: Time
+    integer, intent(in), dimension(4) :: axes
+    integer :: io, ierr, nml_unit, stdlog_unit
+    character(len=32) :: method_str = ''
+
+#ifdef INTERNAL_FILE_NML
+    read(input_nml_file, nml=large_scale_cloud_nml, iostat=io)
+    ierr = check_nml_error(io, 'large_scale_cloud_nml')
+#else
+    if (file_exist('input.nml')) then
+      nml_unit = open_namelist_file()
+      ierr = 1
+      do while (ierr /= 0)
+        read(nml_unit, nml=large_scale_cloud_nml, iostat=io, end=10)
+        ierr = check_nml_error(io, 'large_scale_cloud_nml')
+      enddo
+10    call close_file(nml_unit)
+    endif
+#endif
+    stdlog_unit = stdlog()
+    write(stdlog_unit, large_scale_cloud_nml)
+
+    ! Select cloud fraction diag formula
+    method_str = uppercase(trim(cf_diag_formula_name))
+
+    if(method_str == 'SPOOKIE') then
+      cf_diag_formula = B_SPOOKIE
+      call error_mesg(mod_name, 'Using default SPOOKIE cloud fraction diagnostic formula.', NOTE)
+
+    else if(method_str == 'SUNDQVIST') then
+      cf_diag_formula = B_SUNDQVIST
+      call error_mesg(mod_name, 'Using Sundqvist (1989) cloud fraction diagnostic formula.', NOTE)
+
+    else if(method_str == 'LINEAR') then
+      cf_diag_formula = B_LINEAR
+      call error_mesg(mod_name, 'Using linear cloud fraction diagnostic formula.', NOTE)
+
+    else if(method_str == 'SMITH') then
+      cf_diag_formula = B_SMITH
+      call error_mesg(mod_name, 'Using Smith (1990) cloud fraction diagnostic formula.', NOTE)
+
+    else if(method_str == 'SLINGO') then
+      cf_diag_formula = B_SLINGO
+      call error_mesg(mod_name, 'Using Slingo (1980) cloud fraction diagnostic formula.', NOTE)
+
+    else if(method_str == 'XR96') then
+      cf_diag_formula = B_XR96
+      call error_mesg(mod_name, 'Using Xu and Krueger (1996) cloud fraction diagnostic formula.', NOTE)
+
+    else
+      call error_mesg(mod_name, '"'//trim(cf_diag_formula_name)//'"'// &
+                ' is not a valid cloud fraction diagnostic formula.', FATAL)
+    endif
+
+    if (cf_diag_formula .eq. B_SPOOKIE .or. cf_diag_formula .eq. B_SUNDQVIST &
+    .or. cf_diag_formula .eq. B_SMITH) then
+      do_simple_rhcrit = .true.
+    else
+      do_simple_rhcrit = .false.
+    end if
+
+    if(do_simple_rhcrit) then
+      id_rhcrit = register_diag_field (mod_name, 'rhcrit', axes(1:3), Time, &
+                    'critical relative humidity', '%')
+    end if
+
+  end subroutine large_scale_cloud_init
+
+  subroutine large_scale_cloud_diag(pfull, ps, rh, q_hum, qsat, qcl_rad, wg_full, cf, Time)
```

Comment:
I found the '_diag' routine name a bit misleading. In this context it means diagnosing clouds in the column, but it reads as though we are initialising diagnostics or similar. If it is not too much of a pain, could we change this to 'large_scale_cloud'